cmd/kubeadm/app/phases/certs/pkiutil/pki_helpers.go

```diff
@@ -0,0 +1,157 @@
+	}
+
+	config := certutil.Config{
+		CommonName: "kubernetes-client",
+	}
+	cert, err := certutil.NewSignedCert(config, key, caCert, caKey)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to sign certificate [%v]", err)
+	}
+
+	return cert, key, nil
+}
+
+func WriteCertAndKey(pkiPath string, name string, cert *x509.Certificate, key *rsa.PrivateKey) error {
+	certificatePath, privateKeyPath := pathsForCertAndKey(pkiPath, name)
+
+	if key != nil {
+		if err := certutil.WriteKey(privateKeyPath, certutil.EncodePrivateKeyPEM(key)); err != nil {
+			return fmt.Errorf("unable to write private key to file %q: [%v]", privateKeyPath, err)
+		}
+	}
+
+	if cert != nil {
+		if err := certutil.WriteCert(certificatePath, certutil.EncodeCertPEM(cert)); err != nil {
+			return fmt.Errorf("unable to write certificate to file %q: [%v]", certificatePath, err)
+		}
+	}
+
+	return nil
+}
+
+func TryLoadCertAndKeyFromDisk(pkiPath, name string) (*x509.Certificate, *rsa.PrivateKey, bool, error) {
+	certificatePath, privateKeyPath := pathsForCertAndKey(pkiPath, name)
+	// Suppose one of the cert and the key exist
+	oneOfThemExists := true
+
+	_, certErr := os.Stat(certificatePath)
+	_, keyErr := os.Stat(privateKeyPath)
+	if os.IsNotExist(certErr) && os.IsNotExist(keyErr) {
+		// The cert and the key did not exist, let the caller know that none of them exist
+		oneOfThemExists = false
+		return nil, nil, oneOfThemExists, fmt.Errorf("the certificate or key doesn't exist")
+	}
+
+	certs, err := certutil.CertsFromFile(certificatePath)
+	if err != nil {
+		return nil, nil, oneOfThemExists, fmt.Errorf("couldn't load the certificate file %s: %v", certificatePath, err)
+	}
+
+	// We are only putting one certificate in the certificate pem file, so it's safe to just pick the first one
+	cert := certs[0]
+
+	// Parse the private key from a file
+	privKey, err := certutil.PrivateKeyFromFile(privateKeyPath)
+	if err != nil {
+		return nil, nil, oneOfThemExists, fmt.Errorf("couldn't load the private key file %s: %v", privateKeyPath, err)
+	}
+	var key *rsa.PrivateKey
+	switch k := privKey.(type) {
+	case *rsa.PrivateKey:
+		key = k
+	case *ecdsa.PrivateKey:
+		// TODO: Abstract rsa.PrivateKey away and make certutil.NewSignedCert accept a ecdsa.PrivateKey as well
+		// After that, we can support generating kubeconfig files from ecdsa private keys as well
+		return nil, nil, oneOfThemExists, fmt.Errorf("the private key file %s isn't in RSA format", privateKeyPath)
+	default:
+		return nil, nil, oneOfThemExists, fmt.Errorf("the private key file %s isn't in RSA format", privateKeyPath)
+	}
+
+	// Check so that the certificate is valid now
+	now := time.Now()
+	if now.Before(cert.NotBefore) {
+		return nil, nil, oneOfThemExists, fmt.Errorf("the certificate is not valid yet")
+	}
+	if now.After(cert.NotAfter) {
+		return nil, nil, oneOfThemExists, fmt.Errorf("the certificate is has expired")
+	}
+
+	return cert, key, oneOfThemExists, nil
```

Comment:
oneOfThemExists should just be cert != nil XOR key != nil. The return arg is redundant and unnecessary.